### PR TITLE
Internal: Added AI tests [ED-11035]

### PR DESCRIPTION
**Pull request by KingYes**

### CODE REVIEW

File: tests/playwright/sanity/modules/ai/ai-text.test.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+				if ( requestPostData.includes( 'ai_get_user_information' ) ) {
+					await route.fulfill( {
+						json: userInformationMock,
+					} );
+				}
+			} );
+
+			await expect( page.getByText( 'Use text' ) ).toHaveCount(1 );
+
+			await page.locator( 'button[aria-label="close"]' ).click();
```
Comment (TzviRabinovitch) [inline tests/playwright/sanity/modules/ai/ai-text.test.js:49]: maybe extract this to a function in the editor-page.js

File: tests/playwright/sanity/modules/ai/user-information.mock.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+		success: true,
+		data: {
+			responses: {
+				ai_get_user_information: data,
+			}
+		}
+	}
+}
+
+export const userInformationMock = getJsonMock( {
```
Comment (TzviRabinovitch) [inline tests/playwright/sanity/modules/ai/user-information.mock.js:12]: maybe extract to another file or an external resouces

File: tests/playwright/sanity/modules/ai/user-information.test.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	test( 'User Information', async ( { page }, testInfo ) => {
+		const wpAdmin = new WpAdminPage( page, testInfo );
+
+		const editor = await wpAdmin.openNewPage();
+		await editor.addWidget( 'heading' );
+
+		await test.step( 'Connected and Get started', async () => {
+			await page.click( '.e-ai-button' );
+
+			await page.route( '/wp-admin/admin-ajax.php', async ( route ) => {
```
Comment (TzviRabinovitch) [inline tests/playwright/sanity/modules/ai/user-information.test.js:20]: maybe extract all AI Routs to a function in the editor or somewhere else?

**Review by TzviRabinovitch — CHANGES_REQUESTED** [review]

File: tests/playwright/sanity/modules/ai/ai-button.test.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+test.describe( 'AI @ai', () => {
+	test( 'AI Button', async ( { page }, testInfo ) => {
+		const wpAdmin = new WpAdminPage( page, testInfo );
+
+		const editor = await wpAdmin.openNewPage();
+
+		await test.step( 'Textarea control', async () => {
+			await editor.addWidget( 'heading' );
+
+			await expect( page.locator( '.elementor-control-title.elementor-control-type-textarea .e-ai-button' ) ).toHaveCount(1 );
```
Comment (dmaziar) [inline tests/playwright/sanity/modules/ai/ai-button.test.js:13]:
```
	const widgets = [ 'heading', 'text-editor', 'image'];
			for ( const i in widgets ) {
				const id = await editorPage.addWidget( widgets[ i ] );
				await expect( page.getByRole( 'button', { name: 'Write with AI' } ) ).toHaveCount( 1 );
				await editorPage.removeElement( id );
			}

```

File: tests/playwright/sanity/modules/ai/ai-button.test.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+		await test.step( 'Media control', async () => {
+			await editor.addWidget( 'image' );
+
+			await expect( page.locator( '.elementor-control-image.elementor-control-type-media .e-ai-button' ) ).toHaveCount(1 );
+		} );
+
+		await test.step( 'Media control', async () => {
+			await editor.addWidget( 'image' );
+
+			await page.click( '.elementor-panel-navigation-tab.elementor-tab-control-advanced' );
```
[the remark this answers is not in the document]
Comment (dmaziar) [reply tests/playwright/sanity/modules/ai/ai-button.test.js:31]: I would move it to a separate test.

File: tests/playwright/sanity/modules/ai/ai-button.test.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+			await page.click( '.elementor-control-_section_background.elementor-control-type-section' );
+			await page.locator( '.elementor-control-_background_background.elementor-control-type-choose label[data-tooltip="Classic"]' ).click();
```
Comment (dmaziar) [inline tests/playwright/sanity/modules/ai/ai-button.test.js:33]: await page.locator('[data-tooltip="Classic"]').click() will do the same

File: tests/playwright/sanity/modules/ai/ai-button.test.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+			await expect( page.locator( '.elementor-control-image.elementor-control-type-media .e-ai-button' ) ).toHaveCount( 1 );
+		} );
+
+		await test.step( 'Media control', async () => {
+			await editor.addWidget( 'image' );
+
+			await editor.activatePanelTab( 'advanced' );
+			await editor.openSection( '_section_background' );
+			await page.locator(  '.elementor-control-_background_background .eicon-paint-brush' ).click();
```
Comment (louiswol94) [inline tests/playwright/sanity/modules/ai/ai-button.test.js:33]:
```suggestion
			await page.locator( '.elementor-control-_background_background .eicon-paint-brush' ).click();
```

File: package.json [hunk 8]
```diff
@@ -21,6 +21,7 @@
         "test:playwright:nested-tabs": "npx cross-env TEST_SUITE=@nested-tabs playwright test -c tests/playwright/playwright.config.js",
         "test:playwright:container": "npx cross-env TEST_SUITE=@container playwright test -c tests/playwright/playwright.config.js",
         "test:playwright:nested-accordion": "npx cross-env TEST_SUITE=@nested-accordion playwright test -c tests/playwright/playwright.config.js",
+		"test:playwright:ai": "npx cross-env TEST_SUITE=@ai playwright test -c tests/playwright/playwright.config.js",
```
[the remark this answers is not in the document]
Comment (KingYes) [reply package.json:24]: I fixed all this file to tabs.

**Review by TzviRabinovitch — APPROVED** [review]

**Pull request merged**